samples/pipeline/unit-of-work/Core_6/Endpoint/Program.cs

```diff
@@ -0,0 +1,137 @@
+using System;
+using System.Threading.Tasks;
+using NServiceBus;
+using NServiceBus.Pipeline;
+
+class Program
+{
+    static void Main()
+    {
+        AsyncMain().GetAwaiter().GetResult();
+    }
+
+    static async Task AsyncMain()
+    {
+        Console.Title = "Samples.UnitOfWork.Endpoint";
+        var endpointConfiguration = new EndpointConfiguration("Samples.UnitOfWork.Endpoint");
+        endpointConfiguration.UsePersistence<InMemoryPersistence>();
+        endpointConfiguration.EnableInstallers();
+        endpointConfiguration.SendFailedMessagesTo("error");
+
+        endpointConfiguration.RegisterComponents(r => r.ConfigureComponent(b => new IMySession(), DependencyLifecycle.InstancePerUnitOfWork));
+
+        var endpointInstance = await Endpoint.Start(endpointConfiguration)
+            .ConfigureAwait(false);
+
+        var key = default(ConsoleKeyInfo);
+
+        Console.WriteLine("Press any key to send messages, 'q' to exit");
+        while (key.KeyChar != 'q')
+        {
+            key = Console.ReadKey();
+
+            for (var i = 0; i < 4; i++)
+            {
+                await endpointInstance.SendLocal(new MyMessage());
+            }
+        }
+
+        await endpointInstance.Stop()
+            .ConfigureAwait(false);
+    }
+}
+
+class MyUowBehavior : Behavior<IIncomingPhysicalMessageContext>
```

Comment:
Shouldn't the behavior be stateless and not use DI? Then within the Invoke method, use context.Builder to build the UoW-scoped session.